detectron2/utils/events.py

```diff
@@ -478,9 +497,282 @@ def clear_images(self):
         """
         self._vis_data = []
 
+    def clear_predictions(self):
+        """
+        Delete all predictions from `predictions` list.
+        """
+        self._predictions = []
+
     def clear_histograms(self):
         """
         Delete all the stored histograms for visualization.
         This should be called after histograms are written to tensorboard.
         """
         self._histograms = []
+
+
+class WandbWriter(EventWriter):
+    """
+    Write all scalars to a wandb tool.
+    """
+
+    def __init__(self, cfg: CfgNode, window_size: int = 20, **kwargs):
+        """
+        Args:
+            cfg (CfgNode): the project level configuration object
+            window_size (int): the scalars will be median-smoothed by this window size
+
+            kwargs: other arguments passed to `wandb.init(...)`
+        """
+        self._window_size = window_size
+        self._media = []
+        self.cfg = cfg
+        self.thing_class_names = []
+        self.thing_index_to_class = []
+        self.stuff_class_names = []
+        self.stuff_index_to_class = []
+        self._table_thing_classes = [] # List[wandb.Classes] objects needed for tables
+        self._table_stuff_classes = [] 
+        self._evalset_table = []
```

Comment:
`self.stuff` - does this need to be renamed?

Reply:
I don't understand. Why should it be renamed?